docs/userGuide/syntax/diagrams.mbdf

```diff
@@ -0,0 +1,144 @@
+## Diagrams
```

Comment:
Should we move it under plugins instead? (since it is a plugin after all) #1151 

I'm thinking of removing the site/page nav from `usingComponents` as well (in a separate pr), since it dosen't really fit there with the rest of the vue components, and is already repeated under `tweaking the page structure`

Reply:
In the UG we should go by what things look like to users, rather than how they are implemented. They should appear under plugins only if users need to manipulate them as plugins. Similarly, _components_ in the UG doesn't mean Vue components but building blocks readers would be inserting into pages.

Repetition is not an issue as MarkBind is specifically designed to handle repetition. The idea is, things should appear in _every_ place the reader can expect to see it. Having said that, a good policy is to make such repeated content as 'embedded panels' in all places it is repeated except in one place so that reader can see it supposedly belongs in one place but has been repeated in other places for readers' convenience.

Reply:
ah I see, reason being a new user might not realise it is a plugin before visiting the plugins page, and may mistake installing java/graphviz as a requirement markbind.
How about collapsing the one here into a minimized panel, and inserting something similar to `Setting Site Properties`? (`More info about the site.json file can be found in the panel below.` -> `You can include plantUml diagrams via the plantUml plugin as well... click the panel below to find out more)

Reply:
For the site/page nav, we can make it an embed in a separate pr on that note

Reply:
> ah I see, reason being a new user might not realise it is a plugin before visiting the plugins page, and may mistake installing java/graphviz as a requirement markbind.
> How about collapsing the one here into a minimized panel, and inserting something similar to `Setting Site Properties`? (`More info about the site.json file can be found in the panel below.` -> `You can include plantUml diagrams via the plantUml plugin as well... click the panel below to find out more)

Note that I've removed all mentions of _internal plugins_ from the UG because the user has no need to know about them. As _PlantUML plugin_ is an internal plugin, users don't even know that it is a plugin.

Hence, from the user POV, PlantUML support is a MarkBind feature. If the user wants to use plantUML diagrams, he has to install java etc. and it _is_ a MarkBind requirement. 

Related: My initial idea was not to have _any_ plugins as I wanted _everything built-in_ to be one of our differentiating factors. But we ended up with some plugins along the way. Still, I want to minimize the amount of plugins-related stuff users need to deal with.

Reply:
Ahh ok noted. Also just realised it would only break if you actually use a `<puml>` tag, otherwise not having graphviz is fine.